add include and exclude patterns to jte-model (#243 Part 2)

chkl opened the pull request:
As suggested in #243

casid commented:
Looks good to me! @edward3h, you know the model extension a lot better, I'd say you should have the final say before merge :-)

As with the other PR, maybe we could add a test for this, possible integration test? There are `test/jte-runtime-cp-test-models`(for Maven) and `test/jte-runtime-cp-test-models-gradle` (for Gradle), which could configured with such patterns?

kelunik commented on jte-models/src/main/java/gg/jte/models/generator/ModelExtension.java:
```diff
@@ -31,6 +33,15 @@ public JteExtension init(Map<String, String> value) {
+
+        var templateDescriptionsFiltered = templateDescriptions.stream() //
+                .filter(x -> includePattern == null || includePattern.matcher(x.fullyQualifiedClassName()).matches()) //
+                .filter(x -> excludePattern == null || excludePattern.matcher(x.fullyQualifiedClassName()).matches()) //
```
```suggestion
                .filter(x -> excludePattern == null || ! excludePattern.matcher(x.fullyQualifiedClassName()).matches()) //
```

kelunik commented:
Also, this PR includes both changes now. We should either remove them or close #244.

chkl commented:
> Also, this PR includes both changes now. We should either remove them or close #244.

Thanks @kelunik for reviewing it and good catch. I added the missing negation and dropped the commits that are already included in #244 .


I'm going to add tests for this and #244 later.
